docs/data/data-grid/layout/MinMaxHeightGrid.js

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import Stack from '@mui/material/Stack';
+import { DataGrid } from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+import Typography from '@mui/material/Typography';
+import { useResizeObserver } from '@mui/x-internals/useResizeObserver';
+
+const minHeight = 200;
+const maxHeight = 400;
+
+export default function MinMaxHeightGrid() {
+  const [nbRows, setNbRows] = React.useState(3);
+  const removeRow = () => setNbRows((x) => Math.max(0, x - 1));
+  const addRow = () => setNbRows((x) => Math.min(100, x + 1));
+  const containerRef = React.useRef(null);
+
+  const { data, loading } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: 100,
+    maxColumns: 6,
+  });
+
+  return (
+    <Box sx={{ width: '100%' }}>
+      <Stack direction="row" spacing={1} sx={{ mb: 1 }}>
+        <Button size="small" onClick={removeRow}>
+          Remove a row
+        </Button>
+        <Button size="small" onClick={addRow}>
+          Add a row
+        </Button>
+      </Stack>
+      <div style={{ position: 'relative' }} ref={containerRef}>
+        <div
+          style={{
+            display: 'flex',
+            flexDirection: 'column',
+            maxHeight,
+            minHeight,
+          }}
+        >
+          <DataGrid {...data} rows={data.rows.slice(0, nbRows)} loading={loading} />
+        </div>
+
+        {/** Visualize max and min container height */}
+        <ContainerMeasurements containerRef={containerRef} />
+      </div>
+    </Box>
+  );
+}
+
+function ContainerMeasurements({ containerRef }) {
+  const [containerHeight, setContainerHeight] = React.useState(0);
+
+  useResizeObserver(containerRef, (entries) => {
```

Comment:
I added this component to visualize the DataGrid height (no need to check it in the dev tools):

<img width="677" src="https://github.com/user-attachments/assets/fb7c4f81-d010-43c8-8a95-bf04b14cf31a">

It works fine locally, but the resize observer callback is not called in the deploy preview for some reason 🫠
I'll look into it later, but the PR in its current form is ready for review.

Reply:
Fixed by [`14cbefa` (#14614)](https://github.com/mui/mui-x/pull/14614/commits/14cbefa7ed55269cf30ee8a06830ca37a250c145)